lightning/src/ln/channel.rs

```diff
@@ -1821,7 +1821,33 @@ where
 				res
 			},
 			ChannelPhase::Funded(mut funded_channel) => {
-				let res = funded_channel.commitment_signed(msg, logger).map(|monitor_update_opt| (None, monitor_update_opt));
+				#[cfg(splicing)]
+				let has_negotiated_pending_splice = funded_channel.pending_splice.as_ref()
+					.map(|pending_splice| pending_splice.funding.is_some())
+					.unwrap_or(false);
+				#[cfg(splicing)]
+				let session_received_commitment_signed = funded_channel
+					.interactive_tx_signing_session
+					.as_ref()
+					.map(|session| session.has_received_commitment_signed());
+				#[cfg(splicing)]
+				let res = if has_negotiated_pending_splice
+					// Not having a signing session implies they've already sent `splice_locked`,
+					// which must always come after the initial commitment signed is sent.
+					&& !session_received_commitment_signed.unwrap_or(true)
+				{
+					funded_channel
+						.splice_initial_commitment_signed(msg, logger)
+						.map(|monitor_update_opt| (None, monitor_update_opt))
+				} else {
```

Comment:
We'll hit this when `has_negotiated_pending_splice && session_received_commitment_signed` are true when we'd be expecting a `batch`. But I guess the checks in `FundedChannel::commitment_signed` will catch this. Where refactoring `pending_splice` and `pending_funding`, I can see if we'd want to re-organize any of that logic.